source/_posts/2016-03-23-rake-db-migrate-vs-rake-db-schema-load.markdown

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+title: "rake db:schema:load vs rake db:migrate"
+date: 2016-03-23 08:09:25 +0100
+comments: true
+author: Ivica Lakatoš
+categories:
+  - Ruby and Rails
+tags:
+  - ruby
+  - rails
+  - rake task
+  - globalize
+published: true
+---
+
+Sooner or later every new Ruby developer needs to understand  differences between this two common rake tasks. Basically, a simple definition tells us everything we need to know:
+
++ `rake db:migrate` runs migrations that have not run yet
++ `rake db:schema:load` loads the schema.db file into database.
+
+but the right question is when to use one or the other.
+
+**Advice:** <a id="advice"></a> when you adding new migration to a existing app then you need to run `rake db:migrate`, but when you join to existing application (_especially some old application_), or when you drop your applications database and you need to create it again, always run `rake db:schema:load` to load schema.
+
+### Example
+
+I am working on application which use [globalize gem](https://github.com/globalize/globalize) for ActiveRecord model/data translations. Globalize work this way:
+
++ first specify attributes which need to be translatable
+
+``` ruby
+class Post < ActiveRecord::Base
+  translates :title, :text
+end
+```
+<!--more-->
+
++ then create translation tables
+
+```
+class CreatePosts < ActiveRecord::Migration
+  def up
+    create_table :posts do |t|
+      t.timestamps
+    end
+    Post.create_translation_table! title: :string, text: :text
+  end
+  def down
+    drop_table :posts
+    Post.drop_translation_table!
+  end
+end
+```
+**Note** _that the ActiveRecord model Post must already exist and have listed attributes for translations_.
+
++ and run `rake db:migrate` .
+
+Problem come when you change your mine and deside to leave title to be untranslatable. 
```

Comment:
@ikera I think that here you should have "Problem come**s** when you change your min**d**.."